cmd/soci/commands/create.go

```diff
@@ -79,6 +81,16 @@ var CreateCommand = cli.Command{
 		}
 		spanSize := cliContext.Int64(spanSizeFlag)
 		minLayerSize := cliContext.Int64(minLayerSizeFlag)
+		// Creating the snapshotter's root path first if it does not exist, since this ensures, that
+		// it has the limited permission set as drwx--x--x.
+		// The subsequent oci.New creates a root path dir with too broad permission set.
+		if _, err := os.Stat(config.SociSnapshotterRootPath); os.IsNotExist(err) {
+			if err = os.Mkdir(config.SociSnapshotterRootPath, 0711); err != nil {
+				return err
+			}
+		} else if err != nil {
```

Comment:
Interesting. I didn't know that variable assignments in an if statement are still bound in else statements. I think this would still be clearer if we assigned the error variable in a separate line, though.